### DISCUSSION TITLE
Unable to properly install kafkabridge crd

### DISCUSSION THREAD

**Discussion by Zia-** [strimzi/strimzi-kafka-bridge#788] [Q&A] [answered]
Hey guys,

I assume this is the best place to seek some help regarding properly using _kafkabridge_ in minikube cluster.

To give you the specs, I'm new to the _Kafka_ and _k8s_ world; and currently using _Minikube_ for all testing purposes. I have used the _Minikube_ tutorial from this page https://strimzi.io/quickstarts to get going, and it all works like a charm. However, at this stage, I want to expose the _Kafka_ server to the host machine (the machine where _Minikube_ is running as a container). For this, _kafkabridge_ would be needed as explained here https://strimzi.io/blog/2019/07/19/http-bridge-intro; also because I want to communicate via HTTP protocol.

While applying _kafka-bridge.yaml_, I got the following error message:

```
error: resource mapping not found for name: "my-bridge" namespace: "" from "kafka_bridge.yaml": no matches for kind  

"KafkaBridge" in version "kafka.strimzi.io/v1alpha1"

ensure CRDs are installed first
```

From what I can see, _kafkabridge_ crd needs to be installed first. Chatgpt suggested me to install it using the following command:

`kubectl apply -f 'https://strimzi.io/install/latest?namespace=kafka&crd=kafka-bridge'`

This apparently installed crd, which I verified by running _kubectl get crds_ command:

`kafkabridges.kafka.strimzi.io         2023-04-19T16:37:27Z`

After updating my _kafka_brdige.yaml_ to edit the apiVersion line to:

`apiVersion: kafkabridges.kafka.strimzi.io/v1alpha1`

I'm still getting the following error msg:

```
error: resource mapping not found for name: "my-bridge" namespace: "" from "kafka_bridge.yaml": no matches for kind 

"KafkaBridge" in version "kafkabridges.kafka.strimzi.io/v1alpha1"

ensure CRDs are installed first
```

At this stage, I don't know what's missing. I'm very close to setting up a working kafka for other services. I would appreciate any sort of help in this regard. Let me know if more info is needed to debug the scenario. 

Note: all _kubectl_ commands were run inside _kafka_ namespace.

**Comment by scholzj** [comment, marked as answer]
Well, ChatGTP was wrong and right at the same time I guess. That is not the way how you install the CRD => the `&crd=kafka-bridge` is completely made up. But it is not enough to install a single CRD. You have to install the whole operator, which is done by this command `kubectl apply -f 'https://strimzi.io/install/latest?namespace=kafka`. So it was lucky.

The error you are getting is because you are using an old version of the custom resource => this is because you copied it from a blog post from 2019 ;-). We have up to date examples on GitHub (https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/bridge/kafka-bridge.yaml) and in the docs. Or in the case of the Bridge, you can also just edit the YAML file and use `kafka.strimzi.io/v1beta2` instead of `kafka.strimzi.io/v1alpha1`. That should I think allow you to apply your YAML.

**Reply by Zia-** [reply]
Thanks a lot @scholzj. Learning all these concepts as I'm writing :)
